protocol/nats_jetstream/v2/options.go

```diff
@@ -9,9 +9,14 @@ import (
 	"errors"
 
 	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
 )
 
 var ErrInvalidQueueName = errors.New("invalid queue name for QueueSubscriber")
+var ErrNoConsumerConfig = errors.New("no consumer config was given")
```

Comment:
Question: do we need all those typed errors? They're now part of the API. Also, in those cases a more simple pattern is to use an error struct, for example https://github.com/aws-controllers-k8s/eventbridge-controller/blob/67a9375a6ac7f95f1a3eb8b1b49f329755305c24/pkg/resource/rule/hooks.go#L29-L43

Reply:
Actually, with [`errors.As`](https://pkg.go.dev/errors#As), using simple exported errors works rather than needing to do something more complicated unless you want additional methods.